state-chain/pallets/cf-trading-strategy/src/lib.rs

```diff
@@ -529,6 +670,94 @@ impl<T: Config> Pallet<T> {
 			}) >= FixedU64::one(),
 		)
 	}
+
+	/// Logic for one side of the inventory-based strategy.
+	///
+	/// Given the amount of asset on this side compared to the total amount on both sides, returns
+	/// the limit orders that should be created. The logic is as follows:
+	/// If there is too much asset on this side, ie amount > half_total, then we want 2 limit
+	/// orders:
+	/// 1. A simple order at the average tick between min_tick and max_tick, with half of the total
+	///    amount.
+	/// 2. A dynamic order at a tick that is more aggressive than the average tick with the
+	///    remaining amount.
+	///
+	/// The reason for splitting the order in 2 is to avoid having all of the asset at the most
+	///    aggressive tick get executed and become an order on the opposite side also at the most
+	///    aggressive tick, stuck in a non-profitable loop. By splitting the order at the half
+	///    total we maximize the chance of the strategy balancing out to 50/50 over time.
+	///
+	/// If there is not too much asset on this side, ie amount <= half_total, then we want 1 limit
+	/// order:
+	/// 1. A dynamic order at a tick that is more defensive than the average tick. This is the same
+	///    logic as the dynamic order above.
+	fn inventory_based_strategy_logic(
+		amount: AssetAmount,
+		total: AssetAmount,
+		min_tick: Tick,
+		max_tick: Tick,
+		side: Side,
+	) -> BTreeMap<Tick, (OrderId, AssetAmount)> {
+		if total == 0 {
+			return BTreeMap::default();
+		}
+		let mut orders = BTreeMap::new();
+
+		let fraction_of_total = if side == Side::Sell {
```

Comment:
nit: we don't need this in this scope. It can be defined where we calculate the dynamic tick.

I also (subjectively) find that inverting the fraction is not very intuitive. We could keep the fraction as-is, and then in the sell case, we add it to the min tick, and in the buy case we subtract from the max tick.